src/engine/source/builder/builders/opBuilderHelperFilter.cpp

```diff
@@ -1479,4 +1479,254 @@ base::Expression opBuilderHelperIsFalse(const std::string& targetField,
         });
 }
 
+//*************************************************
+//*              Definition filters               *
+//*************************************************
+
+// <field>: +match_value/$<definition_array>|$<array_reference>
+base::Expression opBuilderHelperMatchValue(const std::string& targetField,
+                                           const std::string& rawName,
+                                           const std::vector<std::string>& rawParameters,
+                                           std::shared_ptr<defs::IDefinitions> definitions)
+{
+    auto parameters {helper::base::processParameters(rawName, rawParameters, definitions, false)};
+    helper::base::checkParametersSize(rawName, parameters, 1);
+
+    const auto name = helper::base::formatHelperName(rawName, targetField, parameters);
+
+    std::optional<std::vector<json::Json>> definitionArray;
+
+    if (helper::base::Parameter::Type::VALUE == parameters[0].m_type)
+    {
+        // Parameter is a definition
+        json::Json definitionValue {};
+
+        try
+        {
+            definitionValue = json::Json {parameters[0].m_value.c_str()};
+        }
+        catch (std::runtime_error& e)
+        {
+            throw std::runtime_error(
+                fmt::format("Engine builder: [{}] Definition '{}' has an invalid type", name, parameters[0].m_value));
+        }
+
+        definitionArray = definitionValue.getArray();
+        if (!definitionArray.has_value())
+        {
+            throw std::runtime_error(
+                fmt::format("Engine builder: [{}] Definition '{}' is not an array", name, parameters[0].m_value));
+        }
+    }
+
+    // Tracing
+    const std::string successTrace {fmt::format("[{}] -> Success", name)};
+
+    const std::string failureTrace1 {fmt::format("[{}] -> Failure: Target field '{}' not found", name, targetField)};
+    const std::string failureTrace2 {
+        fmt::format("[{}] -> Failure: Target field '{}' has an invalid type", name, targetField)};
+    const std::string failureTrace3 {
+        fmt::format("[{}] -> Failure: Parameter '{}' not found", name, parameters[0].m_value)};
+    const std::string failureTrace4 {
+        fmt::format("[{}] -> Failure: Parameter '{}' is not an array", name, parameters[0].m_value)};
+    const std::string failureTrace5 {
+        fmt::format("[{}] -> Failure: Array '{}' does not contain '{}'", name, parameters[0].m_value, targetField)};
+
+    // Return Term
+    return base::Term<base::EngineOp>::create(
+        name,
+        [=,
+         targetField = std::move(targetField),
```

Comment:
why copy the traces string but move the targetField string?
I think we should unify